packages/mco/components/discovered-application-wizard/wizard-steps/configuration-step/recipe-selection.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import { useACMSafeFetch } from '@odf/mco/hooks';
+import { SearchResultItemType } from '@odf/mco/types';
+import { queryRecipesFromCluster } from '@odf/mco/utils';
+import { StatusBox, useCustomTranslation } from '@odf/shared';
+import { SingleSelectDropdown } from '@odf/shared/dropdown';
+import { getName } from '@odf/shared/selectors';
+import { getValidatedProp } from '@odf/shared/utils';
+import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+import { FormGroup, SelectOption, Text } from '@patternfly/react-core';
+import {
+  EnrollDiscoveredApplicationAction,
+  EnrollDiscoveredApplicationStateType,
+  NAMESPACE_NAME_SPLIT_CHAR,
+} from '../../utils/reducer';
+import './configuration-step.scss';
+
+const getNamespaceName = (name: string, namespace: string) =>
+  // Multiple namespace can have recipe with same name,
+  // Converting into namespace_name for unique identification in dropdown option
+  !!name && !!namespace
+    ? `${name}${NAMESPACE_NAME_SPLIT_CHAR}${namespace}`
```

Comment:
nits: if we are storing name + namespace then variable should be named in same order to avoid unnecessary confusion...
```suggestion
const getNameNamespace = (name: string, namespace: string) =>
  // Multiple namespace can have recipe with same name,
  // Converting into namespace_name for unique identification in dropdown option
  !!name && !!namespace
    ? `${name}${NAME_NAMESPACE_SPLIT_CHAR}${namespace}`
```